cordial_behavior/scripts/behaviors.py

```diff
@@ -0,0 +1,260 @@
+			self._result.interaction_continue = INTERACTION_CONTINUE
+			self._result.interacting_action = goal_name
+			self._result.message = INTERACTION_MESSAGE
+			TRACKER_STATE = ''
+			TRACKER_DONE = False
+			DETECTOR_MESSAGE = ''
+			self.action.set_succeeded(self._result)
+
+
+class BehaviorServer():
+	_feedback = InteractionFeedback()
+	_result = InteractionResult()
+	def __init__(self, name):
+		self.action_name = name
+		self.action = actionlib.SimpleActionServer(self.action_name, InteractionAction, self.execute_goal, False)
+		self.action.start()
+
+	def execute_goal(self, goal):
+		global TTS_MESSAGE, SPEAKER_DONE
+		goal_name = goal.interacting_action
+		success = True
+		if goal.optional_data != '':
+			TTS_MESSAGE = goal.optional_data
+		bm = BehaviorManager()
+		bm.handle_behavior(TTS_MESSAGE)
+		self._feedback.interacting_action = goal_name
+		self._feedback.interaction_state = FEEDBACK_MESSAGE
+		## Decide when to send the feedback
+		# self.action.publish_feedback(self._feedback)
+		while not SPEAKER_DONE:
+			if self.action.is_preempt_requested():
+					self.action.set_preempted()
+					success = False
+			rospy.Rate(10)
+		if success:
+			self._result.interaction_continue = INTERACTION_CONTINUE
+			self._result.interacting_action = goal_name
+			self._result.message = INTERACTION_MESSAGE
+			SPEAKER_DONE = False
+			TTS_MESSAGE = ''
+			self.action.set_succeeded(self._result)
+		
+
+class BehaviorManager():
+
+	def __init__(self):
+		rospy.Subscriber('cordial/behavior', Behavior, self.handle_tts_message)
+		rospy.Subscriber('cordial/speaker/done', Bool, self.handle_speaker_message)
+		rospy.Subscriber('cordial/detector/faces', String, self.handle_detector_message) # I dont know if it is a string!
+		rospy.Subscriber('cordial/behavior/tracking/state', String, self.handle_tracking_state) # States: NO_TRACK, ACQUIRING, FOUND, LOST 
+		self.face_publisher = rospy.Publisher('cordial/behavior/face/expressing', FaceRequest, queue_size=10)
+		self.gesture_publisher = rospy.Publisher('cordial/behavior/gesture/moving', Gesture, queue_size=10)
+		self.speaker_publisher = rospy.Publisher('cordial/speaker/playing', PlayRequest, queue_size=10)
+		self.tracker_publisher = rospy.Publisher('cordial/behavior/tracking', Bool, queue_size=1)
+		self.get_facial_expressions_list()
+		#self.handle_behavior(data) #FOR TESTING
+		
+
+	def handle_tracking_state(self, data):
+		global TRACKER_STATE
+		TRACKER_STATE = data.data
+
+	def handle_tts_message(self, data):
+		global TTS_MESSAGE
+		TTS_MESSAGE = data
+		print("The message received from the TTS is:", TTS_MESSAGE)
+
+			
+	def handle_speaker_message(self, data):
+		global SPEAKER_DONE
+		SPEAKER_DONE = data.data
+		rospy.loginfo("The speaker has finished: " +str(data.data))
+
+	def handle_detector_message(self, data):
+		global DETECTOR_MESSAGE
+		DETECTOR_MESSAGE = data
+
+	def handle_tracker(self, data):
+		self.tracker_publisher.publish(True)
+
+		
+
+	def get_facial_expressions_list(self):
```

Comment:
so this gets the list of possible facial expressions and also stores their keyframes? I think this makes sense, but I think it might change slightly based on having parameterized behaviors--In which case, the keyframe au_degrees and au_ms become strings that represent functions to be evaluated later.

I think this is fine, but I wanted to note this for later